### ISSUE TITLE
Add the `recovery` pallet to the Polkadot runtime

### ISSUE THREAD

**Issue by Tbaut** [polkadot-fellows/runtimes#96]
This pallet has been present on Kusama for many years, and is currently under-utilized but extremely important IMHO.
https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/recovery

If the appropriate tooling is built around this pallet, it would bring a lot of safety to users assets. Allowing them to effectively recover funds, NFT etc, in case they loose access to their accounts. I believe that it is currently not widely supported in the ecosystem wallets because it is only present on Kusama, hence this issue to bring it to the Polkadot runtime.

**Comment by bkchr** [comment]
I support this. The pallet should go onto some system parachain on Polkadot. I would probably choose collectives, but maybe @joepetrowski has a better idea?

**Comment by joepetrowski** [comment]
Either Collectives or the new People Chain. It would probably take a little work to make it recover balances cross-chain, but it'd be a nice feature. As in, when you recover an account on the chain with the Recovery pallet, you can send a message to transfer balance with the origin of the recovered account on other chains.

**Comment by bkchr** [comment]
> It would probably take a little work to make it recover balances cross-chain, but it'd be a nice feature.

You mean it would require some wallet support to achieve this? Or what is the issue here? Using the current implementation you should be already able to send a XCM message?

**Comment by joepetrowski** [comment]
Send a message, yes, but recovery just sets the recoverer [as a proxy](https://github.com/paritytech/polkadot-sdk/blob/878c562/substrate/frame/recovery/src/lib.rs#L610), and proxies are not recognized cross-chain. You can send a message via proxy but the XCM Executor will add a `ClearOrigin` and you won't be able to withdraw funds. Need to check a few things but it will be possible, we can write an XCM program to transfer funds to the new account.

**Comment by xlc** [comment]
It will take quite amount of work to make this use case working with XCM but it is super useful and we should do it. And I think people chain will be the good place.

**Comment by bkchr** [comment]
@franciscoaguirre can you help here with the XCM side?

**Comment by franciscoaguirre** [comment]
Yes, I'll be looking into it

**Comment by seadanda** [comment]
Happy to work on this from the People Chain side

**Comment by bkchr** [comment]
BTW I just thought about this today. Isn't the recovery pallet just a glorified multisig with a time delayed proxy? Aka we should be able to build this already, without requiring the pallet at all? The only thing missing would be the xcm support as well for this.

Do I miss something? WDYT @shawntabrizi?

**Comment by joepetrowski** [comment]
Good point, we could do https://github.com/paritytech/polkadot-sdk/issues/85 and then make the XCM changes to set proxies on other chains.

**Comment by bkchr** [comment]
> Placing the deposit to initiate the recovery process is unique and makes it very clear when something is going on.

But you could also track if someone starts a multisig operation?
